SimulationAgent/Agent.cs

```diff
@@ -59,6 +65,16 @@ public async Task RunAsync()
             while (this.running)
             {
                 var oldSimulation = this.simulation;
+                DateTime now = DateTime.Now;
+                TimeSpan duration = now - this.lastPolledTime;
+                
+                // Send heartbeat every 24 hours
+                if (duration.Days >= DIAGNOSTICS_POLLING_FREQUENCY)
+                {
+                    this.lastPolledTime = DateTime.Now;
+                    this.logDiagnostics.LogServiceHeartbeatAsync();
```

Comment:
pls add test for this code. We should verify that the heartbeat is sent only once a day, and not for every loop (which would DDOS the endpoint)